lib/bundler/rubygems_integration.rb

```diff
@@ -795,6 +802,12 @@ def use_gemdeps(gemfile)
         activated_spec_names = runtime.requested_specs.map(&:to_spec).sort_by(&:name)
         [Gemdeps.new(runtime), activated_spec_names]
       end
+
+      if provides?(">= 2.5.2")
```

Comment:
this entire file is "hacks to support 7 years of rubygems versions", in that context nothing is an edge case :P